wheatley/row_generation/complib_composition_generator.py

```diff
@@ -58,27 +59,38 @@ def removeprefix(string: str, prefix: str) -> str:
 class ComplibCompositionGenerator(RowGenerator):
     """ The RowGenerator subclass for generating rows from a CompLib composition. """
 
-    complib_url = "https://complib.org/composition/"
+    complib_url = "https://api.complib.org/composition/"
 
     def __init__(self, comp_id: int, access_key: Optional[str]=None) -> None:
+        # Generate URL and request the rows
         url = self.complib_url + str(comp_id) + "/rows"
         if access_key:
             url += "?accessKey=" + access_key
         request_rows = requests.get(url)
-
+        # Check for the status of the requests
         if request_rows.status_code == 404:
             raise InvalidCompError(comp_id)
-
         if request_rows.status_code == 403:
             raise PrivateCompError(comp_id)
-
         request_rows.raise_for_status()
+        # Parse the request responses as JSON
+        response_rows = json.loads(request_rows.text)
 
         # New line separated, skip the first line (rounds)
-        split_rows = request_rows.text.splitlines(False)[1::]
-        self.loaded_rows: List[Row] = [Row([Bell.from_str(bell) for bell in row]) for row in split_rows]
+        split_rows = response_rows['rows']
+        self.loaded_rows: List[Tuple[Row, Optional[str]]] = [
+            (
+                Row([Bell.from_str(bell) for bell in row]),
+                None if call == '' else call
+            ) for row, call, thing_that_I_dont_know_what_it_does in split_rows
```

Comment:
I believe this is a flag enum, so looking a the binary representation you can read off properties.
e.g. 8 if it's a handstroke, 16 if it's a leadhead, 256 for a leadend (which are also handstrokes so will be at least 256 + 8 = 264)

Reply:
Ah that's cool.  I wonder if it would be worth storing them...? I guess it's probably worth just ignoring them for now.